packages/fx-core/src/common/cpUtils.ts

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { LogProvider } from "@microsoft/teamsfx-api";
+import * as cp from "child_process";
+import * as os from "os";
+
+export async function executeCommand(
+  workingDirectory: string | undefined,
+  logger: LogProvider | undefined,
+  options: cp.SpawnOptions | undefined,
+  command: string,
+  ...args: string[]
+): Promise<string> {
+  const result: ICommandResult = await tryExecuteCommand(
+    workingDirectory,
+    logger,
+    options,
+    command,
+    ...args
+  );
+  if (result.code !== 0) {
+    const errorMessage = `Failed to run command: "${command} ${result.formattedArgs}", code: "${result.code}",
+                            output: "${result.cmdOutput}", error: "${result.cmdOutputIncludingStderr}"`;
+    await logger?.debug(errorMessage);
+    throw new Error(errorMessage);
+  } else {
+    await logger?.debug(`Finished running command: "${command} ${result.formattedArgs}".`);
+  }
+
+  return result.cmdOutput;
+}
+
+async function tryExecuteCommand(
+  workingDirectory: string | undefined,
+  logger: LogProvider | undefined,
+  additionalOptions: cp.SpawnOptions | undefined,
+  command: string,
+  ...args: string[]
+): Promise<ICommandResult> {
+  return await new Promise(
+    (resolve: (res: ICommandResult) => void, reject: (e: Error) => void): void => {
+      let cmdOutput = "";
+      let cmdOutputIncludingStderr = "";
+      const formattedArgs: string = args.join(" ");
+
+      workingDirectory = workingDirectory || os.tmpdir();
+      const options: cp.SpawnOptions = {
+        cwd: workingDirectory,
+        shell: true,
+      };
+      Object.assign(options, additionalOptions);
+
+      const childProc: cp.ChildProcess = cp.spawn(command, args, options);
+      let timer: NodeJS.Timeout;
+      if (options.timeout && options.timeout > 0) {
+        // timeout only exists for exec not spawn
+        timer = setTimeout(() => {
+          childProc.kill();
+          logger?.debug(
+            `Stop exec due to timeout, command: "${command} ${formattedArgs}", options = '${JSON.stringify(
+              options
+            )}'`
+          );
+          reject(
+            new Error(`Exec command: "${command} ${formattedArgs}" timeout, ${options.timeout} ms`)
+          );
+        }, options.timeout);
+      }
+      logger?.debug(
+        `Running command: "${command} ${formattedArgs}", options = '${JSON.stringify(options)}'`
+      );
+
+      childProc.stdout?.on("data", (data: string | Buffer) => {
+        data = data.toString();
+        cmdOutput = cmdOutput.concat(data);
+      });
+
+      childProc.stderr?.on("data", (data: string | Buffer) => {
+        data = data.toString();
+        cmdOutputIncludingStderr = cmdOutputIncludingStderr.concat(data);
```

Comment:
based on the variable name, my understanding is the output here include both stdout and stderr? Is it possible to only provide stderr to consumers? Since there's already a cmdOutput that stores stdout.